src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -6416,8 +6417,9 @@ namespace ZoneTempPredictorCorrector {
                 //  how can we tell?  predict step must be lagged ?  correct step, systems have run.
                 auto const &node(Node(zec.InletNode(NodeNum)));
                 NodeTemp = node.Temp;
+                NodeHumRat = node.HumRat;
                 MassFlowRate = node.MassFlowRate;
-                CpAir = PsyCpAirFnW(ZoneAirHumRat(ZoneNum));
+                CpAir = PsyCpAirFnW(0.5 * (NodeHumRat + ZoneAirHumRat(ZoneNum)));
```

Comment:
Maybe the averaging of w is too simplistic here and there would need to be mass-flow-weighting across all the inlet nodes.  In the context of more than one inlet node, the average humidity ratio may need to account for all the inlet nodes.  One low flow inlet may be doing little to change humidity ratio compared to another high flow inlet.  Consider the aggregate/balance latent load across multiple inlets.

Reply:
I agree, zone entering humidity ratio should be mass flow rate weighted average across all inlet nodes of a zone. Will look into it and make appropriate changes. @EnergyArchmage  Thanks.

Reply:
I understand the intent here of including all of the other nodes, but that's going to make it impossible to match what happens here with the HVAC system output controls that use a similar calculation to figure out how much flow and/or what temperature to supply to meet the zone load. The system only knows about it's supply node, not any others.

Reply:
This is inside a loop over inlets (or ADUs, depending on which loop) so there is already some weighting by flow as `SumSysMCp` and `SumSysMCpT` accumulate within these loops.

Looking further up in this function, infiltration and other flows come in from `MCPI, MCPV` etc. These are calculated in `ZoneAirLoopEquipmentManager::CalcAirFlowSimple` And these are not consistent. Infiltration and ventilation use the source cp, mixing uses the average, refrigeration door mixing uses both zones (somewhow).

For the zone balance, using the source humrat for Cp seems reasonable, because you're looking at the energy to move the source air to the zone condition. Granted it should be the zone condition after correction, but we don't have the luxury of knowing that.